src/runtime/driver/handle.rs

```diff
@@ -0,0 +1,286 @@
+//! Internal, reference-counted handle to the driver.
+//!
+//! The driver was previously managed exclusively by thread-local context, but this proved
+//! untenable.
+//!
+//! The new system uses a handle which reference-counts the driver to track ownership and access to
+//! the driver.
+//!
+//! There are two handles.
+//! The strong handle is owning, and the weak handle is non-owning.
+//! This is important for avoiding reference cycles.
+//! The weak handle should be used by anything which is stored in the driver or does not need to
+//! keep the driver alive for it's duration.
+
+use io_uring::{cqueue, squeue};
+use std::cell::RefCell;
+use std::io;
+use std::os::unix::io::{AsRawFd, RawFd};
+use std::rc::{Rc, Weak};
+use std::task::{Context, Poll};
+
+use crate::buf::fixed::FixedBuffers;
+use crate::runtime::driver::op::{Completable, Lifecycle, MultiCQEFuture, Op, Updateable};
+use crate::runtime::driver::Driver;
+
+#[derive(Clone)]
+pub(crate) struct Handle {
+    pub(super) inner: Rc<RefCell<Driver>>,
```

Comment:
> How would you feel about naming this field driver? And the WeakHandle's inner `driver`` too? Just thinking out loud. This isn't even a nit. Just wanting to discuss names, the crate will live with them a long time.

I think that this is a good idea.

> Also where the Handle type is used for other fields in this PR, maybe name the field handle instead of driver? There's some precedent for that in the tokio crate I think. Then one can say or think "the handle to the driver", and it syncs with the code. I guess thinking "the weak handle to the driver" would mean the inner field below could be named driver too.

I named it driver because we might end up with multiple handles potentially. I could see us having handles for things like buffer registries as well.

Reply:
I think the Handle for the driver will be the most important so it would be okay to go by the short name handle. If there are other Handle types down the road, and the variable is long lived, it gets a name other than simply handle. JMO.